compiler/src/iree/compiler/Codegen/LLVMGPU/VectorReductionToGPU.cpp

```diff
@@ -82,23 +83,40 @@ static Value groupReduction(Location loc, OpBuilder &builder, Value input,
   return laneVal;
 }
 
-/// Special case to hoist hal operations that have side effect but are safe to
-/// move out of the warp single lane region.
-static void hoistHalBindingOps(vector::WarpExecuteOnLane0Op warpOp) {
+/// Hoist uniform operations as well as special hal operations that have side
+/// effect but are safe to move out of the warp single lane region.
+void static moveScalarAndBindingUniformCode(
```

Comment:
TIL `void static` is a thing. Interesting.

Reply:
oops... changing